7_drivers_munge/src/GCM_driver_nc_utils.R

```diff
@@ -3,25 +3,99 @@
+  }
 
-  # TODO: remove this step which just creates an empty file
-  # with a .nc extension as a placeholder
-  writeLines("PLACEHOLDER", nc_file)
+  # Read in all data for the current (tar_group) GCM
+  gcm_data <- purrr::map_df(gcm_raw_files, function(gcm_raw_file) {
+    arrow::read_feather(gcm_raw_file)
+  }) %>% arrange(cell)
+
+  # Pull vector of unique dates and convert to POSIXct
+  gcm_dates <- as.POSIXct(unique(gcm_data$time), tz='GMT')
+
+  # Pull vector of unique cell numbers
+  gcm_cells <- unique(gcm_data$cell)
+  gcm_cells_dim_name = "gcm_cell_id"
+
+  # Get WGS84 latitude and longitude of cell centroids
+  cell_coords <- spatial_info %>%
+    mutate(lon = sf::st_coordinates(.)[,1], lat = sf::st_coordinates(.)[,2]) %>%
+    rename(cell = cell_no) %>%
+    sf::st_set_geometry(NULL) %>%
+    arrange(cell) %>%
+    filter(cell %in% gcm_cells)
+
+  # Get vectors of cell centroid latitudes and longitudes
+  cell_centroid_lats <- cell_coords %>% pull(lat)
+  cell_centroid_lons <- cell_coords %>% pull(lon)
+
+  # Set up attributes for NetCDF that are independent of variables
+  data_time_units <- "days since 1970-01-01 00:00:00"
+  data_attributes <- list('title' = global_att)
+  data_attributes <- append(data_attributes, grid_params) # add grid parameters to list of global attributes
+  data_coordvar_long_names <- list(instance = "identifier for GCM grid cell", time = "date",
+                              lat = "WGS84 latitude of GCM grid cell centroid", lon = "WGS84 longitude of GCM grid cell centroid",
+                              alt = "NULL")
+
+  # Pivot data to long format to set up for filtering by variable
+  gcm_data_long <- gcm_data %>%
+    pivot_longer(cols = -c(time, cell), names_to = "variable", values_to = "value") %>%
+    arrange(cell)
+
+  # Loop over data variables to populate NetCDF using ncdfgeom::write_timeseries_dsg()
+  for (variable in vars_info$var_name) {
+    # Filter long-format data to selected variable
+    variable_metadata <- vars_info %>% filter(var_name == variable)
+    var_data <- gcm_data_long %>%
+      filter(variable == !!variable) %>%
+      pivot_wider(id_cols = c("time"), names_from = cell, values_from = value) %>%
+      select(-time) %>%
+      as.data.frame() %>%
+      setNames(gcm_cells)
+
+    # Pull attributes for selected variable
+    var_data_unit <- rep(variable_metadata$units, length(gcm_cells))
+    var_data_prec <- variable_metadata$precision
+    var_data_metadata <- list(name = variable_metadata$var_name, long_name = variable_metadata$longname)
+
+    # Check if nc file already exists, and therefore should be added to,
+    # or if it needs to be created
+    var_add_to_existing <- ifelse(file.exists(nc_file), TRUE, FALSE)
+
+    # write this variable to the netcdf file:
+    # ncdfgeom::write_timeseries_dsg() function documentation:
+    # https://github.com/USGS-R/ncdfgeom/blob/master/R/write_timeseries_dsg.R
+    ncdfgeom::write_timeseries_dsg(nc_file,
```

Comment:
One think to think about when arranging the data is how it is likely to be accessed in chunks, or contiguous parts of the `gcm_cells` instances. It should be some kind of logical arrangement, like left to right, bottom to top. I see you've done that here:

![image](https://user-images.githubusercontent.com/2349007/151453549-7e80f25f-e2e9-4fc0-905d-5dfd30429959.png)
(redder are instances closer to the beginning of the vector)
This looks good to me, but putting this check in so you can see it. I used my old-school
```r
plot(cell_centroid_lons, cell_centroid_lats, col = colorRampPalette(c("red", "blue"))(length(cell_centroid_lats)))
```